outlay/app/src/test/java/app/outlay/example/CategoriesPresenterTest.java

```diff
@@ -0,0 +1,78 @@
+package app.outlay.example;
+
+import app.outlay.core.executor.DefaultSubscriber;
+import app.outlay.domain.interactor.GetCategoriesUseCase;
+import app.outlay.domain.interactor.UpdateCategoriesUseCase;
+import app.outlay.domain.model.Category;
+import app.outlay.mvp.presenter.CategoriesPresenter;
+import app.outlay.mvp.view.CategoriesView;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by MR.ESSIG on 5/2/2017.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CategoriesPresenterTest {
+
+    private CategoriesPresenter categoriesPresenter;
+    private List<Category> categories;
+
+    @Mock
+    private GetCategoriesUseCase getCategoriesUseCaseMock;
+
+    @Mock
+    private UpdateCategoriesUseCase updateCategoriesUseCaseMock;
+
+    @Mock
+    private CategoriesView categoriesViewMock;
+
+    @Captor
+    private ArgumentCaptor<DefaultSubscriber<List<Category>>> CategoriesSubscriberCaptor;
+
+    @Captor
+    private ArgumentCaptor<DefaultSubscriber> SubscriberCaptor;
+
+
+    @Before
+    public void init(){
+        categoriesPresenter = new CategoriesPresenter(getCategoriesUseCaseMock,updateCategoriesUseCaseMock);
+        categoriesPresenter.attachView(categoriesViewMock);
+        categories = new ArrayList<>();
+    }
+
+    @Test
+    public void getCategories_UsesGetCategoriesUseCase(){
+        categoriesPresenter.getCategories();
+
+        verify(getCategoriesUseCaseMock).execute(any());
+    }
+
+    @Test
+    public void getCategories_DefaultSubscriber_ShowsViewCategories_OnNext(){
+        categoriesPresenter.getCategories();
+
+        verify(getCategoriesUseCaseMock).execute(CategoriesSubscriberCaptor.capture());
+        CategoriesSubscriberCaptor.getValue().onNext(categories);
+        verify(categoriesViewMock).showCategories(categories);
```

Comment:
What we're testing here is whether the presenter passes an empty `ArrayList` to the view. This test wouldn't detect if it always passed an empty list to the view, instead of the data it received from the callback. I suggest adding at least one `Category` to the list.